### ISSUE TITLE
Instance segmentation active learning: Ignoring corrupt image/label: cannot reshape array of size

### ISSUE THREAD

**Issue by jinyoonok2** [ultralytics/ultralytics#1742]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

**Hello ultralytics team, Here is explanation about the problem I am currently facing**

**Environment:**
Google Colab
**Set up:** 
Ultralytics YOLOv8.0.28 🚀 Python-3.9.16 torch-1.13.1+cu116 CUDA:0 (Tesla T4, 15102MiB)
Setup complete ✅ (4 CPUs, 25.5 GB RAM, 25.8/166.8 GB disk)

**What I am trying to do:**
What I am trying to do is **active learning** that trains over partial data for the "segment" operation yolov8
This first "partial data" will be labeled with hands but I want the trained model to apply instance segmentation for the leftovers.

for the first yolo command, I used this for train
**!yolo
task=segment
mode=train
model=yolov8s-seg.pt data=/content/drive/MyDrive/Study/datasets/tomato-train-100-3/data.yaml
epochs=20
imgsz=256**

now I got the weight for new best.pt file, I am going to use it to predict the leftover images to give them labels
and I will use them again for the next training step.

**yolo
task=segment
mode=predict
model=(my directory)/YOLOv8/runs/segment/train/weights/best.pt
conf=0.50 source=/content/drive/MyDrive/Study/datasets/tomato-active-learning/Tomato-Early-blight-leaf
save=true
save_txt=true
save_conf=true**

since I set save_txt=true, these will output label images on predict

take these as labels and take the leftover images that I previously prepared as the images for the next training.

I collected the labels from:
/YOLOv8/runs/segment/predict/labels
/YOLOv8/runs/segment/predict 2/labels
/YOLOv8/runs/segment/predict 3/labels
...

now changed their names fit to the category: Tomato-Early-blight-leaf_(number)
(ex: Tomato-Early-blight-leaf_1.txt, Tomato-Early-blight-leaf_2.txt, ....)

and then split them to put them into train/valid directories.

-tomato-active-learning
  -data.yaml
  -train
    -images
      -Tomato-Early-blight-leaf_1.jpg <--these are the images that I used for the prediction, now they will be used for training with the generated label from them.
      -Tomato-Early-blight-leaf_2.jpg
      -Tomato-Early-blight-leaf_3.jpg
      -...
    -labels
      -Tomato-Early-blight-leaf_1.txt
      -Tomato-Early-blight-leaf_2.txt
      -Tomato-Early-blight-leaf_3.txt
      -...
  -valid
      -images
      -Tomato-Early-blight-leaf_111.jpg
      -Tomato-Early-blight-leaf_112.jpg
      -Tomato-Early-blight-leaf_113.jpg
      -...
    -labels
      -Tomato-Early-blight-leaf_111.txt
      -Tomato-Early-blight-leaf_112.txt
      -Tomato-Early-blight-leaf_113.txt
      -...

Structure is basicly like this.
the content of the data.yaml is below:

path: /content/drive/MyDrive/Study/datasets/tomato-active-learning
train: ./train/images
val: ./valid/images
names:
  0: Tomato-Early-blight-leaf
  1: Tomato-Septoria-leaf-spot
  2: Tomato-leaf
  3: Tomato-leaf-bacterial-spot
  4: Tomato-leaf-late-blight
  5: Tomato-leaf-mosaic-virus
  6: Tomato-leaf-target-spot
  7: Tomato-leaf-yellow-virus
  8: Tomato-mold-leaf
  9: Tomato-two-spotted-spider

and then I get the following errors
it seems like it does not recognize the label files(I also deleted label.cache file before every start)

![errormessage](https://user-images.githubusercontent.com/84653106/229249412-b3f12380-8bfe-4d87-8e19-7df954157298.png)

**what have I tried:**

1.
First of all, I wanted to check whether the label file is really corrupted or not
so I tried to upload sample images and labels to the roboflow and I could check that
the label file is fit for the image file and they are printing fine.

example in roboflow:
![examplerobo](https://user-images.githubusercontent.com/84653106/229250285-eeeed71e-f3e1-4dfd-8eff-0a58f74254fd.png)

Why am I not using the roboflow then?
Because the number of data I am trying to use is huge(10000+) roboflow does not accept
every data I have in their space.
Therefore, I was trying to do the custom data with manual set up.

2.
I tried to change the data yaml format
such like
-Tomato-Early-blight-leaf
-Tomato-Septoria-leaf-spot
-Tomato-leaf
-Tomato-leaf-bacterial-spot
...

or something like this

[Tomato-Early-blight-leaf, Tomato-Septoria-leaf-spot, ...]

3.
I checked the directory path again,
I filtered the images that were not annotated by the previous predict,
I also checked the labels that were predicted wrong and changed them to the correct class.
ex: in label folder, suppose there is a Tomato-Early-blight-leaf that is labeled in the wrong class
    9 0.864062 0.00625 0.8625 0.0078125 0.853125 0.0078125 ...
->  0 0.864062 0.00625 0.8625 0.0078125 0.853125 0.0078125 ...

After all these tries, I am still getting the same error message.

4.
Is the label data normalied?
As we can see from above example, the label data is perfectly normalized and range between 0 and 1

**In conclusion, what I want:**
I train the yolo model with the partially hand labeled set of data,
With that trained model, I want to generate the label data (instance segment) with predict from remaining data,
collaborate them with the remaing data,
then train the model again with the generated label + remaining image.

and it does not seem to recognize the labels to the images in the predict process.
May I please get help for analysis of the problem?

**Comment by glenn-jocher** [comment]
👋 Hello @Cirakoglu, thank you for your detailed explanation of the problem you are facing. It looks like you're trying to use custom data and train it through active learning for instance segmentation with YOLOv8. 

From the error message you received, it seems like YOLOv8 could not find the label files for the validation set. Make sure the annotated label files are located in the correct folder (`./valid/labels` in your case) and that they have the same name as the corresponding image files, except for the extension (`.txt` instead of `.jpg`, for example).

Additionally, if you modified your data YAML file, make sure that the folder paths and class IDs are updated correctly to match the new data structure.

I hope this helps! Please let me know if you have any further questions.

**Comment by jinyoonok2** [comment]
Thank you for your reply!

I tried changing some path in data.yaml file and it still seems not to work so well as I intended
Below is the command line I used to train for the step of active learning

%cd {HOME}
!yolo
task=segment
mode=train
model=/content/drive/MyDrive/Study/PV_Project/YOLOv8/runs/segment/train2/weights/best.pt data=/content/drive/MyDrive/Study/datasets/tomato-active-learning/data.yaml
epochs=10
imgsz=256

above yaml file is in the directory beside the "train" and "valid" folders that have dataset in them.
the structure is like the one from the previous post

-tomato-active-learning
  **-data.yaml**
  **-train**
    -images
      -Tomato-Early-blight-leaf_1.jpg
      -Tomato-Early-blight-leaf_2.jpg
      -Tomato-Early-blight-leaf_3.jpg
      -...
    -labels
      -Tomato-Early-blight-leaf_1.txt
      -Tomato-Early-blight-leaf_2.txt
      -Tomato-Early-blight-leaf_3.txt
      -...
  **-valid**
    -images
      -Tomato-Early-blight-leaf_111.jpg
      -Tomato-Early-blight-leaf_112.jpg
      -Tomato-Early-blight-leaf_113.jpg
      -...
    -labels
      -Tomato-Early-blight-leaf_111.txt
      -Tomato-Early-blight-leaf_112.txt
      -Tomato-Early-blight-leaf_113.txt
      -...


and I tried to change the content of the yaml files like the following:

**path: /content/drive/MyDrive/Study/datasets/tomato-active-learning
train: ./train
val: ./valid**
nc: 10
names:
  0: Tomato-Early-blight-leaf
  1: Tomato-Septoria-leaf-spot
  2: Tomato-leaf
  3: Tomato-leaf-bacterial-spot
  4: Tomato-leaf-late-blight
  5: Tomato-leaf-mosaic-virus
  6: Tomato-leaf-target-spot
  7: Tomato-leaf-yellow-virus
  8: Tomato-mold-leaf
  9: Tomato-two-spotted-spider


I also tried changing them to:

**train: ./train/images
val: ./valid/images**

it seems the both way are not working.
at the start, the yolo seem to recognize my settings like the following image:
![workingfineatstart](https://user-images.githubusercontent.com/84653106/229317132-cfd86f3e-c99d-4e1c-b3df-626eab087b7c.png)

but when it gets to match the images and the labels, it does not recognize my labels and skip every images before it finds no matching labels when there are text labels folder right next to the images folder.

![errorattheend](https://user-images.githubusercontent.com/84653106/229317351-7444ebe4-15f0-4e6b-bb2d-2ebadfcd16d3.png)

As I checked at the roboflow in the previous post, the robofllow only can recognize the label text files and the image files match when their names are the same.

I am uncertain with that my yaml file structure is not correct.
Would there be anything I can do to fix this error furthermore? So far, I tried but could not find the way to solve the problem.

**Comment by glenn-jocher** [comment]
@Cirakoglu, thank you for the additional information. Based on the command line you provided, it seems that the path to the model weights (`/content/drive/MyDrive/Study/PV_Project/YOLOv8/runs/segment/train2/weights/best.pt`) is different from the path to your data YAML file (`/content/drive/MyDrive/Study/datasets/tomato-active-learning/data.yaml`). Make sure that both paths are correct and match the locations of the actual files.

Regarding the structure of your YAML file, it appears to be correct based on the information you provided. However, I would suggest checking your label files to make sure they are formatted correctly. The label file should have a line for each object in the image, in the following format:

`<class_id> <x_center> <y_center> <width> <height>`

For example:

`3 0.488183 0.506250 0.179699 0.227083`

Make sure that the `class_id` corresponds to the correct class in your `names` dictionary in the YAML file.

If your label files are formatted correctly and located in the correct folder (`./valid/labels`), then YOLOv8 should be able to find and use them during training/validation.

I hope this helps! Let me know if you have any further questions.

**Comment by jinyoonok2** [comment]
Hello, Thank you for your replies!

**(1)**
I was putting data.yaml file and best.pt file separately because I was trying to place the dataset folder separately on the different location so I could arrange them among other datasets I have. but keep the weight file(best.pt) stay in the YOLOv8 directory.

I am worried if that causes error so I changed the location of the best.pt file at the same location of data.yaml file in case it really works

like below:

!yolo
task=segment
mode=train
model=/content/drive/MyDrive/Study/datasets/tomato-active-learning/best.pt data=/content/drive/MyDrive/Study/datasets/tomato-active-learning/data.yaml
epochs=10
imgsz=640

but it seems getting the same error as before.

**(2)**
I checked the label file content which is like the one you suggested
<class_id> <x_center> <y_center> <width> <height>

it seems to have <class_id> at the front, but it does not seem to have <x_center> <y_center> <width> <height> because it is in polynomial instance segmentation.

Therefore it has multiple coordinates of the locations like below example
Tomato-Early-blight-leaf_1.txt:
**0** 0.864062 0.00625 0.8625 0.0078125 0.853125 0.0078125 0.851562 0.009375 0.846875 0.009375 0.845312 0.0109375 0.84375 0.0109375 0.840625 0.0140625 0.839063 0.0140625 0.829687 0.0234375 0.828125 0.0234375 0.826563 0.025 0.825 0.025 0.823438 0.0265625 0.821875 0.0265625 0.820312 0.028125 0.81875 0.028125 0.817187 0.0296875 0.815625 0.0296875 0.814062 0.03125 0.810938 0.03125 0.809375 0.0328125 0.807813 0.0328125 0.80625 0.034375 0.803125 0.034375 0.801562 0.0359375 0.798437 0.0359375 0.796875 0.0375 0.790625 0.0375 0.789062 0.0390625 0.784375 0.0390625 0.782812 0.040625 0.775 0.040625 0.773438 0.0421875 0.767187 0.0421875 0.765625 0.04375 0.760938 0.04375 0.759375 0.0453125 0.75625 0.0453125 0.754687 0.046875 0.753125 0.046875 0.751562 0.0484375 0.746875 0.0484375 0.745313 0.05 0.74375 0.05 0.742188 0.0515625 0.739062 0.0515625 0.7375 0.053125 0.735937 0.053125 0.734375 0.0546875 0.73125 0.0546875 0.729688 0.05625 0.728125 0.05625 0.726562 0.0578125 0.723437 0.0578125 0.721875 0.059375 0.720312 0.059375 0.71875 0.0609375 0.715625 0.0609375 0.714063 0.0625 0.710938 0.0625 0.709375 0.0640625 0.70625 0.0640625 0.704687 0.065625 0.701563 0.065625 0.7 0.0671875 0.696875 0.0671875 0.695312 0.06875 0.69375 0.06875 0.692187 0.0703125 0.690625 0.0703125 0.689062 0.071875 0.6875 0.071875 0.685938 0.0734375 0.684375 0.0734375 0.682813 0.075 0.68125 0.075 0.678125 0.078125 0.676562 0.078125 0.675 0.0796875 0.673437 0.0796875 0.671875 0.08125 0.670313 0.08125 0.66875 0.0828125 0.667188 0.0828125 0.664062 0.0859375 0.6625 0.0859375 0.660937 0.0875 0.659375 0.0875 0.65625 0.090625 0.654688 0.090625 0.653125 0.0921875 0.651563 0.0921875 0.65 0.09375 0.648438 0.09375 0.646875 0.0953125 0.645312 0.0953125 0.642187 0.0984375 0.640625 0.0984375 0.6375 0.101562 0.635938 0.101562 0.632812 0.104687 0.63125 0.104687 0.628125 0.107813 0.626562 0.107813 0.623438 0.110937 0.621875 0.110937 0.61875 0.114063 0.617188 0.114063 0.6125 0.11875 0.610937 0.11875 0.607813 0.121875 0.60625 0.121875 0.601562 0.126563 0.6 0.126563 0.596875 0.129688 0.595312 0.129688 0.5875 0.1375 0.585938 0.1375 0.578125 0.145313 0.576563 0.145313 0.571875 0.15 0.570312 0.15 0.565625 0.154687 0.564062 0.154687 0.5625 0.15625 0.560938 0.15625 0.557813 0.159375 0.55625 0.159375 0.553125 0.1625 0.551562 0.1625 0.55 0.164062 0.548437 0.164062 0.545313 0.167187 0.54375 0.167187 0.5375 0.173438 0.535937 0.173438 0.528125 0.18125 0.526563 0.18125 0.509375 0.198437 0.507812 0.198437 0.501562 0.204688 0.5 0.204688 0.498437 0.20625 0.496875 0.20625 0.49375 0.209375 0.492188 0.209375 0.490625 0.210938 0.4875 0.210938 0.485938 0.2125 0.482812 0.2125 0.48125 0.214062 0.478125 0.214062 0.476562 0.215625 0.471875 0.215625 0.470313 0.217187 0.465625 0.217187 0.464063 0.21875 0.4625 0.21875 0.460938 0.220313 0.457812 0.220313 0.454688 0.223438 0.453125 0.223438 0.451562 0.225 0.45 0.225 0.448438 0.226562 0.446875 0.226562 0.442187 0.23125 0.440625 0.23125 0.43125 0.240625 0.43125 0.242188 0.425 0.248437 0.425 0.25 0.420312 0.254687 0.420312 0.25625 0.417188 0.259375 0.417188 0.260938 0.415625 0.2625 0.415625 0.264062 0.414062 0.265625 0.414062 0.267188 0.4125 0.26875 0.4125 0.270312 0.410937 0.271875 0.410937 0.273438 0.409375 0.275 0.409375 0.276563 0.407813 0.278125 0.407813 0.279687 0.40625 0.28125 0.40625 0.282813 0.404687 0.284375 0.404687 0.2875 0.403125 0.289062 0.403125 0.290625 0.401563 0.292188 0.401563 0.295312 0.4 0.296875 0.4 0.3 0.398438 0.301562 0.398438 0.303125 0.396875 0.304688 0.396875 0.30625 0.395312 0.307813 0.395312 0.309375 0.39375 0.310937 0.39375 0.3125 0.392188 0.314063 0.392188 0.315625 0.390625 0.317187 0.390625 0.31875 0.3875 0.321875 0.3875 0.323438 0.385938 0.325 0.385938 0.326562 0.379687 0.332812 0.379687 0.334375 0.370313 0.34375 0.370313 0.345313 0.367188 0.348437 0.367188 0.35 0.3625 0.354688 0.3625 0.35625 0.357812 0.360938 0.357812 0.3625 0.35625 0.364062 0.35625 0.365625 0.353125 0.36875 0.353125 0.370313 0.35 0.373437 0.35 0.375 0.346875 0.378125 0.346875 0.379687 0.340625 0.385938 0.340625 0.3875 0.332812 0.395312 0.332812 0.396875 0.321875 0.407813 0.320312 0.407813 0.314063 0.414062 0.3125 0.414062 0.309375 0.417188 0.307813 0.417188 0.298438 0.426562 0.296875 0.426562 0.289062 0.434375 0.289062 0.435937 0.270312 0.454688 0.26875 0.454688 0.253125 0.470313 0.251563 0.470313 0.248437 0.473437 0.246875 0.473437 0.245312 0.475 0.24375 0.475 0.240625 0.478125 0.239063 0.478125 0.2375 0.479688 0.235938 0.479688 0.234375 0.48125 0.232812 0.48125 0.223438 0.490625 0.221875 0.490625 0.21875 0.49375 0.21875 0.495313 0.215625 0.498437 0.215625 0.5 0.204688 0.510938 0.204688 0.5125 0.2 0.517187 0.2 0.51875 0.196875 0.521875 0.196875 0.523438 0.19375 0.526563 0.19375 0.528125 0.192188 0.529688 0.192188 0.53125 0.189063 0.534375 0.189063 0.535937 0.1875 0.5375 0.1875 0.539062 0.185937 0.540625 0.185937 0.542188 0.182812 0.545313 0.182812 0.546875 0.18125 0.548437 0.18125 0.55 0.179688 0.551562 0.179688 0.553125 0.175 0.557813 0.175 0.559375 0.170312 0.564062 0.170312 0.565625 0.167187 0.56875 0.167187 0.570312 0.165625 0.571875 0.165625 0.573438 0.1625 0.576563 0.1625 0.578125 0.157813 0.582812 0.157813 0.584375 0.154687 0.5875 0.154687 0.589063 0.145313 0.598437 0.145313 0.6 0.134375 0.610937 0.134375 0.6125 0.13125 0.615625 0.13125 0.617188 0.126563 0.621875 0.126563 0.623438 0.11875 0.63125 0.11875 0.632812 0.115625 0.635938 0.115625 0.6375 0.1125 0.640625 0.1125 0.642187 0.109375 0.645312 0.109375 0.646875 0.107813 0.648438 0.107813 0.65 0.10625 0.651563 0.10625 0.653125 0.104687 0.654688 0.104687 0.65625 0.103125 0.657812 0.103125 0.659375 0.101562 0.660937 0.101562 0.6625 0.1 0.664062 0.1 0.667188 0.0984375 0.66875 0.0984375 0.670313 0.096875 0.671875 0.096875 0.675 0.0953125 0.676562 0.0953125 0.68125 0.09375 0.682813 0.09375 0.6875 0.0921875 0.689062 0.0921875 0.69375 0.090625 0.695312 0.090625 0.701563 0.0890625 0.703125 0.0890625 0.709375 0.0875 0.710938 0.0875 0.715625 0.0859375 0.717188 0.0859375 0.723437 0.084375 0.725 0.084375 0.729688 0.0828125 0.73125 0.0828125 0.734375 0.08125 0.735937 0.08125 0.740625 0.0796875 0.742188 0.0796875 0.745313 0.078125 0.746875 0.078125 0.75 0.0765625 0.751562 0.0765625 0.753125 0.075 0.754687 0.075 0.75625 0.0734375 0.757812 0.0734375 0.759375 0.071875 0.760938 0.071875 0.7625 0.06875 0.765625 0.06875 0.767187 0.065625 0.770312 0.065625 0.771875 0.0640625 0.773438 0.0640625 0.775 0.0609375 0.778125 0.0609375 0.779688 0.0578125 0.782812 0.0578125 0.784375 0.0546875 0.7875 0.0546875 0.789062 0.0515625 0.792188 0.0515625 0.79375 0.0484375 0.796875 0.0484375 0.798437 0.046875 0.8 0.046875 0.801562 0.0453125 0.803125 0.0453125 0.804688 0.0421875 0.807813 0.0421875 0.809375 0.040625 0.810938 0.040625 0.8125 0.0390625 0.814062 0.0390625 0.815625 0.0375 0.817187 0.0375 0.820312 0.0359375 0.821875 0.0359375 0.823438 0.034375 0.825 0.034375 0.826563 0.0328125 0.828125 0.0328125 0.829687 0.03125 0.83125 0.03125 0.832812 0.0296875 0.834375 0.0296875 0.835938 0.028125 0.8375 0.028125 0.839063 0.0265625 0.840625 0.0265625 0.842188 0.025 0.84375 0.025 0.845312 0.0234375 0.846875 0.0234375 0.85 0.021875 0.851562 0.021875 0.854688 0.0203125 0.85625 0.0203125 0.857813 0.01875 0.859375 0.01875 0.8625 0.0171875 0.864062 0.0171875 0.86875 0.015625 0.870313 0.015625 0.879687 0.0140625 0.88125 0.0140625 0.909375 0.0125 0.910937 0.0125 0.942187 0.0140625 0.94375 0.0140625 0.957812 0.015625 0.959375 0.015625 0.9625 0.0171875 0.964063 0.0171875 0.965625 0.0203125 0.96875 0.0203125 0.970312 0.0234375 0.973437 0.0234375 0.975 0.0265625 0.978125 0.0265625 0.98125 0.028125 0.982813 0.028125 0.984375 0.0296875 0.985937 0.0296875 0.9875 0.0375 0.995313 0.0375 0.998438 0.970312 0.998438 0.970312 0.995313 0.971875 0.99375 0.971875 0.984375 0.973437 0.982813 0.973437 0.976562 0.975 0.975 0.975 0.971875 0.979688 0.967188 0.979688 0.965625 0.98125 0.964063 0.98125 0.9625 0.982813 0.960938 0.982813 0.957812 0.984375 0.95625 0.984375 0.951563 0.985937 0.95 0.985937 0.945312 0.9875 0.94375 0.9875 0.925 0.985937 0.923437 0.985937 0.920313 0.98125 0.915625 0.98125 0.914062 0.979688 0.9125 0.979688 0.910937 0.978125 0.909375 0.978125 0.907812 0.976562 0.90625 0.976562 0.904688 0.975 0.903125 0.975 0.901563 0.973437 0.9 0.973437 0.896875 0.971875 0.895312 0.971875 0.89375 0.970312 0.892187 0.970312 0.890625 0.96875 0.889063 0.96875 0.885938 0.967188 0.884375 0.967188 0.88125 0.965625 0.879687 0.965625 0.878125 0.964063 0.876562 0.964063 0.875 0.960938 0.871875 0.960938 0.870313 0.954687 0.864062 0.954687 0.8625 0.95 0.857813 0.95 0.85625 0.946875 0.853125 0.946875 0.851562 0.945312 0.85 0.945312 0.848437 0.94375 0.846875 0.94375 0.839063 0.942187 0.8375 0.942187 0.825 0.94375 0.823438 0.94375 0.810938 0.945312 0.809375 0.945312 0.803125 0.946875 0.801562 0.946875 0.8 0.948438 0.798437 0.948438 0.795313 0.95 0.79375 0.95 0.790625 0.951563 0.789062 0.951563 0.7875 0.953125 0.785937 0.953125 0.784375 0.954687 0.782812 0.954687 0.78125 0.95625 0.779688 0.95625 0.778125 0.957812 0.776563 0.957812 0.775 0.960938 0.771875 0.960938 0.770312 0.9625 0.76875 0.9625 0.767187 0.964063 0.765625 0.964063 0.745313 0.965625 0.74375 0.965625 0.734375 0.967188 0.732813 0.967188 0.726562 0.96875 0.725 0.96875 0.71875 0.970312 0.717188 0.970312 0.7125 0.971875 0.710938 0.971875 0.709375 0.973437 0.707812 0.973437 0.704687 0.975 0.703125 0.975 0.7 0.976562 0.698438 0.976562 0.69375 0.978125 0.692187 0.978125 0.6875 0.979688 0.685938 0.979688 0.68125 0.98125 0.679688 0.98125 0.673437 0.982813 0.671875 0.982813 0.660937 0.984375 0.659375 0.984375 0.653125 0.985937 0.651563 0.985937 0.648438 0.9875 0.646875 0.9875 0.640625 0.989062 0.639063 0.989062 0.623438 0.990625 0.621875 0.989062 0.620313 0.989062 0.6 0.990625 0.598437 0.990625 0.589063 0.992188 0.5875 0.992188 0.576563 0.99375 0.575 0.99375 0.571875 0.995313 0.570312 0.995313 0.56875 0.996875 0.567187 0.998438 0.567187 0.998438 0.479688 0.996875 0.479688 0.99375 0.476562 0.99375 0.475 0.992188 0.473437 0.992188 0.471875 0.989062 0.46875 0.989062 0.467187 0.982813 0.460938 0.98125 0.460938 0.979688 0.459375 0.978125 0.459375 0.975 0.45625 0.973437 0.45625 0.970312 0.453125 0.96875 0.453125 0.964063 0.448438 0.9625 0.448438 0.960938 0.446875 0.959375 0.446875 0.957812 0.445312 0.95625 0.445312 0.954687 0.44375 0.953125 0.44375 0.95 0.440625 0.948438 0.440625 0.940625 0.432813 0.940625 0.43125 0.939062 0.429688 0.939062 0.428125 0.9375 0.426562 0.9375 0.415625 0.935938 0.414062 0.935938 0.334375 0.934375 0.332812 0.934375 0.320312 0.932813 0.31875 0.932813 0.3125 0.93125 0.310937 0.93125 0.304688 0.929688 0.303125 0.929688 0.296875 0.928125 0.295312 0.928125 0.29375 0.926562 0.292188 0.926562 0.289062 0.925 0.2875 0.925 0.282813 0.923437 0.28125 0.923437 0.278125 0.921875 0.276563 0.921875 0.273438 0.920313 0.271875 0.920313 0.26875 0.91875 0.267188 0.91875 0.264062 0.917188 0.2625 0.917188 0.257812 0.915625 0.25625 0.915625 0.253125 0.914062 0.251563 0.914062 0.248437 0.9125 0.246875 0.9125 0.242188 0.910937 0.240625 0.910937 0.235938 0.909375 0.234375 0.909375 0.23125 0.907812 0.229687 0.907812 0.226562 0.90625 0.225 0.90625 0.220313 0.904688 0.21875 0.904688 0.215625 0.903125 0.214062 0.903125 0.209375 0.901563 0.207813 0.901563 0.201562 0.9 0.2 0.9 0.114063 0.901563 0.1125 0.901563 0.0921875 0.903125 0.090625 0.903125 0.078125 0.904688 0.0765625 0.904688 0.0625 0.90625 0.0609375 0.90625 0.0328125 0.904688 0.03125 0.904688 0.021875 0.903125 0.0203125 0.903125 0.015625 0.901563 0.0140625 0.901563 0.0125 0.9 0.0109375 0.9 0.009375 0.898438 0.0078125 0.89375 0.0078125 0.892187 0.00625 0.942318

I thought this is how it goes for the instance segmentation, it prints perfectly fine when I upload them together with image on the roboflow, but would these multiple coordinates cause errors when I input them into the YOLO model?

**(3)**
train: WARNING ⚠️ .../tomato-active-learning/train/images/Tomato-two-spotted-spider_981.jpg: ignoring corrupt image/label: cannot reshape **array of size 1303 into shape (2)**
train: WARNING ⚠️ .../tomato-active-learning/train/images/Tomato-two-spotted-spider_982.jpg: ignoring corrupt image/label: cannot reshape **array of size 1475 into shape (2)**
train: WARNING ⚠️ .../tomato-active-learning/train/images/Tomato-two-spotted-spider_983.jpg: ignoring corrupt image/label: cannot reshape **array of size 1509 into shape (2)**
train: WARNING ⚠️ .../tomato-active-learning/train/images/Tomato-two-spotted-spider_984.jpg: ignoring corrupt image/label: cannot reshape **array of size 1353 into shape (2)**

so these were the last lines of the error,
I found out from messages that the array size is keep changing.
May I ask why the array size changes over every label?
because I thought this was based on the imgsize of the image,
if every images have same shapes, are array sizes supposed to change?

I am also confused what shape(2) is concerning about,
for example, my images are 256x256 and have 3 channels
Therefore it would be (256,256,3),
Why would it print shape(2) instead of something similar to (256,256,3) like the ones I had read from some other previous issue posts?

May I get explanation about how these label files are accepted by the YOLO models?

I am wondering if there are things that I need to change for the label files for the current progress.

Thank you for your consistent support and time

**Comment by vaskers5** [comment]
Faced with the same problem, I did a little investigation: 

1) Everything is fine with your labels, the fact is that yolo cannot parse your data, here on this [line](https://github.com/ultralytics/ultralytics/blob/a38f22767254f2c2ee241490bd027987780d83ca/ultralytics/yolo/data/utils.py#:~:text=classes%20%3D%20np.array(%5Bx%5B0%5D%20for%20x%20in%20lb%5D%2C%20dtype%3Dnp.float32))

2) I opened this function in the debugger: Problem is here that YOUR Label len is odd, but it's should be divided by two without a remainder.
![image](https://user-images.githubusercontent.com/48238046/232622476-7972bd8f-0e8e-45e9-99e1-2d78ed6111be.png)

FIX: 

Just open your labels and add or delete some points from polygons

**Comment by pderrenger** [comment]
@vaskers5 thank you for sharing your investigation! It seems that the issue is related to the structure of the label files. YOLO requires the annotation format to consist of pairs of coordinates (x, y) for each point, but it seems that your labels contain uneven numbers of coordinates.

To fix this issue, as suggested, you will need to modify your label files to ensure that each instance's annotation consists of even pairs of coordinates (x1, y1, x2, y2, x3, y3, ...). You may need to adjust the annotations to conform to this format.

Once you've adjusted the labels to have an even number of coordinates, YOLO should be able to effectively parse your data without encountering the "cannot reshape array" error.

I hope this helps! If you have any more questions or concerns, feel free to ask.
